src/main/java/com/lucidworks/spark/rdd/SolrRDD.java

```diff
@@ -1,162 +0,0 @@
-package com.lucidworks.spark.rdd;
-
-import com.lucidworks.spark.util.SolrQuerySupport;
-import com.lucidworks.spark.query.ShardSplit;
-import com.lucidworks.spark.query.StreamingResultsIterator;
-import com.lucidworks.spark.query.TermVectorIterator;
-import com.lucidworks.spark.util.SolrSupport;
-import org.apache.log4j.Logger;
-import org.apache.solr.client.solrj.SolrQuery;
-import org.apache.solr.client.solrj.impl.CloudSolrClient;
-import org.apache.solr.client.solrj.response.QueryResponse;
-import org.apache.solr.common.SolrDocument;
-import org.apache.solr.common.params.ModifiableSolrParams;
-import org.apache.spark.SparkContext;
-import org.apache.spark.SparkException;
-import org.apache.spark.api.java.JavaRDD;
-import org.apache.spark.api.java.JavaSparkContext;
-import org.apache.spark.api.java.function.FlatMapFunction;
-import org.apache.spark.mllib.linalg.Vector;
-
-import java.io.Serializable;
-import java.util.Collections;
-import java.util.List;
-
-public class SolrRDD implements Serializable{
```

Comment:
All clients will break with this change ... I don't have any good suggestions how to solve this. One option would be to keep SolrRDD for the Java version and use ScalaSolrRDD for the other one... which is awkward too. Or put them in separate packages - then you could keep the Java version of SolrRDD that extends SolrJavaRDD and mark it as deprecated?